### PR TITLE
votca: tighten up boost dep

### PR DESCRIPTION
**Pull request by snehring**
variants and minimum version from CMakeLists.txt

Upper bounds discovered via trial and error, ran into some issues with boost@1.85 while checking #44776 didn't break the build for this.

### CODE REVIEW

File: var/spack/repos/builtin/packages/votca/package.py [hunk 1]
```diff
@@ -39,7 +39,8 @@ class Votca(CMakePackage):
     depends_on("expat")
     depends_on("fftw-api@3")
     depends_on("eigen@3.3:")
-    depends_on("boost")
+    depends_on("boost@1.71:1.84+filesystem+system+regex+timer")
```
Comment (junghans) [inline var/spack/repos/builtin/packages/votca/package.py]: Why is there an upper bound?

Comment (junghans) [reply var/spack/repos/builtin/packages/votca/package.py]: I thought we fixed that in https://github.com/votca/votca/pull/1123

Comment (snehring) [reply var/spack/repos/builtin/packages/votca/package.py]: boost/filesystem/convenience.hpp was removed in 1.85

Comment (snehring) [reply var/spack/repos/builtin/packages/votca/package.py]: I don't believe that's made it into a release yet. I suppose I should bound it on `@2023:2024`

File: var/spack/repos/builtin/packages/votca/package.py [hunk 2]
```diff
@@ -39,7 +39,9 @@ class Votca(CMakePackage):
     depends_on("expat")
     depends_on("fftw-api@3")
     depends_on("eigen@3.3:")
-    depends_on("boost")
+    depends_on("boost@1.71:+filesystem+system+regex+timer")
+    depends_on("boost@1.71:1.84+filesystem+system+regex+timer", when="@2024")
+    depends_on("boost@1.71:1.82+filesystem+system+regex+timer", when="@:2023")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/votca/package.py]: Can we separate the common requests on boost from the version constraints:
```suggestion
    depends_on("boost+filesystem+system+regex+timer")
    depends_on("boost@1.71:")
    depends_on("boost@:1.84", when="@2024")
    depends_on("boost@:1.82", when="@:2023")
```
?

Comment (snehring) [reply var/spack/repos/builtin/packages/votca/package.py]: Sure. I had the same thought last night, but wasn't sure it'd work that way. Should have tested, but it was late.

File: var/spack/repos/builtin/packages/votca/package.py [hunk 3]
```diff
@@ -39,9 +39,10 @@ class Votca(CMakePackage):
     depends_on("expat")
     depends_on("fftw-api@3")
     depends_on("eigen@3.3:")
-    depends_on("boost@1.71:+filesystem+system+regex+timer")
-    depends_on("boost@1.71:1.84+filesystem+system+regex+timer", when="@2024")
-    depends_on("boost@1.71:1.82+filesystem+system+regex+timer", when="@:2023")
+    depends_on("boost+filesystem+system+regex+timer")
+    depends_on("boost@1.71:")
+    depends_on("boost@1.71:1.84", when="@2024")
```
Comment (junghans) [inline var/spack/repos/builtin/packages/votca/package.py]: I am always confused, does `@2024` means just `v2024` or `v2024.*`? We are about to make the `v2024.1`, which will have the fix to build with boost-1.85.

Comment (snehring) [reply var/spack/repos/builtin/packages/votca/package.py]: I'm regularly tripped up by version specs and non semantic versioning adds on to that, but `@2024` probably would include `2024.1`, so I'll make that restriction exact.

**Review by junghans — APPROVED** [review]

File: var/spack/repos/builtin/packages/votca/package.py (hunk 3 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/votca/package.py]: `@2024` means `2024 <= x < 2025` accoding to our version `<` semantic

**Pull request merged**